src/storage/src/flush/picker.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use async_trait::async_trait;
+use common_telemetry::logging;
+use common_time::util;
+use store_api::logstore::LogStore;
+use store_api::storage::{FlushContext, FlushReason, Region};
+
+use crate::config::DEFAULT_AUTO_FLUSH_INTERVAL;
+use crate::region::RegionImpl;
+
+/// Config for [FlushPicker].
+pub struct PickerConfig {
+    /// Interval to auto flush a region if it has not flushed yet.
+    pub auto_flush_interval: Duration,
+}
+
+impl PickerConfig {
+    /// Returns the interval to pick regions.
+    pub(crate) fn picker_schedule_interval(&self) -> Duration {
+        self.auto_flush_interval / 2
```

Comment:
Yes, you're right. The background flush task is just a supplement to avoid an inactive region or small region not been flushed for a long time. But I think we could expose this config and also set this to a lower value. The largest flush interval is probably `auto_flush_interval + schedule_interval`.